examples/nixnet/flexray-signal-single-point-input.py

```diff
@@ -0,0 +1,138 @@
+ 
+ Running from command line:
+Server machine's IP address, port number, and interface name can be passed as separate command line
+arguments.
+  > python flexray-signal-single-point-input.py <server_address> <port_number> <interface_name>
+If they are not passed in as command line arguments, then by default the server address will be
+"localhost:31763"
+"""
+
+import sys
+
+import grpc
+import nixnet_pb2 as nixnet_types
+import nixnet_pb2_grpc as grpc_nixnet
+
+SERVER_ADDRESS = "localhost"
+SERVER_PORT = "31763"
+INTERFACE = "FlexRay2"
+
+# Read in cmd args
+if len(sys.argv) >= 2:
+    SERVER_ADDRESS = sys.argv[1]
+if len(sys.argv) >= 3:
+    SERVER_PORT = sys.argv[2]
+if len(sys.argv) >= 4:
+    INTERFACE = sys.argv[3]
+
+# Parameters
+DATABASE = "NIXNET_example"
+CLUSTER = "FlexRay_Cluster"
+SIGNAL_LIST = "FlexRayEventSignal1,FlexRayEventSignal2"
+NUM_SIGNALS = 2
+SUCCESS = 0
+
+
+def check_for_error(status):
+    """Raise an exception if the status indicates an error."""
+    if status != 0:
+        error_message_response = client.StatusToString(
+            nixnet_types.StatusToStringRequest(status_id=status)
+        )
+        raise Exception(error_message_response.status_description)
+
+
+i = 0
+keyslot_id = 1
+value_buffer = [0.0] * NUM_SIGNALS
+time_stamp_buffer = [None] * NUM_SIGNALS
+
+# Create the communication channel for the remote host and create connections to the NI-XNET and
+# session services.
+channel = grpc.insecure_channel(f"{SERVER_ADDRESS}:{SERVER_PORT}")
+client = grpc_nixnet.NiXnetStub(channel)
+
+# Display parameters that will be used for the example.
+print("Interface: " + INTERFACE, "Database: " + DATABASE, "Signal List: " + SIGNAL_LIST, sep="\n")
+print("KeySlotId:", keyslot_id)
+
+try:
+    # Create an XNET session in SignalOutSinglePoint mode
+    create_session_response = client.CreateSession(
+        nixnet_types.CreateSessionRequest(
+            database_name=DATABASE,
+            cluster_name=CLUSTER,
+            list=SIGNAL_LIST,
+            interface=INTERFACE,
+            mode=nixnet_types.CREATE_SESSION_MODE_MODE_SIGNAL_IN_SINGLE_POINT,
+        )
+    )
+    check_for_error(create_session_response.status)
+
+    session = create_session_response.session_ref
+    print("Session Created Successfully.\n")
+
+    # Set the Key Slot
+    set_property_response = client.SetProperty(
+        nixnet_types.SetPropertyRequest(
+            session_ref=session,
+            property_id=nixnet_types.PROPERTY_SESSION_INTF_FLEX_RAY_KEY_SLOT_ID,
+            u32_scalar=keyslot_id,
+        )
+    )
+    check_for_error(set_property_response.status)
+
+    while i < 10:
+        # Update the signal data
+        read_signal_response = client.ReadSignalSinglePoint(
+            nixnet_types.ReadSignalSinglePointRequest(
+                session_ref=session,
+                size_of_value_buffer=value_buffer.__sizeof__(),
+                size_of_timestamp_buffer=time_stamp_buffer.__sizeof__(),
```

Comment:
I think we need to wait till tomorrow. This API is being redesigned, so the way it will be used will also change.

Reply:
I didn't see your comment and added a similar comment. I think it should be fine to send this to owners review (with slight change to make it explicit we are passing buffer size in bytes here). Whichever change goes in later should fix the example to change from size-in-bytes to number-of-elements. The field name would change too, so example would anyway have syntax error making it obvious that we need to fix it.